### PR TITLE
Fix bundle report compatibility error

### PR DESCRIPTION
**Pull request by sean-dickinson**
## Description
- Adds use of safe access operator in `GemInfo#state` conditional expression

## Motivation and Context
- This fix is related to issue [#77](https://github.com/fastruby/next_rails/issues/77)
- Without this safe access operator the `else` part of this condition is never reached when `@latest_compatible_version` is `nil`. 

I will say I'm not completely clear on the logic in `#find_latest_compatible`. I don't know enough about the shape of the spec objects and the edge cases surrounding them. It appears to me that the error state is caused when there are specs found by the `Gem::SpecFetcher` but none of them are compatible with the version of Rails given. This will execute `#find_latest_compatible` but never set the `@latest_compatible_version` instance variable. It's a bit confusing as there is a null object pattern being used along with a nil state and it's hard to parse the meaning behind each without more context.

I'd be happy to adjust the solution if this is more of a band-aid than a true fix.

## How Has This Been Tested?
- Running the test suite locally

**I will abide by the [code of conduct](https://github.com/fastruby/next_rails/blob/main/CODE_OF_CONDUCT.md)**

### CODE REVIEW

**Comment by rishijain** [comment]
hey @sean-dickinson thank you for the PR. :heart: 

Do you mind checking the build as it is failing? I think it is due to:
`The safe navigation operator (&.) was added in Ruby 2.3.0.`

So the spec for ruby versions below 2.3.0 are failing. Can you fix those?

File: spec/next_rails/gem_info_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    it "sets latest_compatible_version to NullGem if no specs are found" do
+      fetcher_double = double("spec_fetcher")
+      allow(fetcher_double).to receive(:available_specs).and_return([[],[]])
+      allow(Gem::SpecFetcher).to receive(:new).and_return(fetcher_double)
+      gem = mock_gem.new('gem_name', "0.0.1")
+
+      gem_info = NextRails::GemInfo.new(gem)
+      gem_info.find_latest_compatible
+      expect(gem_info.latest_compatible_version).to be_a(NextRails::GemInfo::NullGemInfo)
+    end
```
[the remark this answers is not in the document]
Comment (sean-dickinson) [reply spec/next_rails/gem_info_spec.rb:55]: Sounds good. Just added it as another spec. I find it a bit awkward with all the mocks, so if you think there's a better way feel free to let me know.

File: lib/next_rails/gem_info.rb [hunk 2]
```diff
@@ -77,14 +77,14 @@ def from_rails?
...
+        :incompatible
       elsif latest_compatible_version.version == "NOT FOUND"
         :no_new_version
-      elsif latest_compatible_version
-        :found_compatible
       else
-        :incompatible
+        :found_compatible
       end
     end
```
Comment (rishijain) [inline lib/next_rails/gem_info.rb:89]: From a visual perspective may be, can we make it like this?

```
      if compatible_with_rails?(rails_version: rails_version)
        :compatible
      elsif latest_compatible_version && latest_compatible_version.version == "NOT FOUND"
        :no_new_version
      elsif latest_compatible_version
        :found_compatible
      else
        :incompatible
      end
    end
```

what do you think @sean-dickinson ?

Comment (sean-dickinson) [reply lib/next_rails/gem_info.rb:89]: Ah yeah that makes more sense than what I did 🙃 . Good call.

**Comment by rishijain** [comment]
Looks good to me @sean-dickinson ! thanks for working on this. I am marking this as approved, but not merging yet for someone from the team to also look at it.

**Review by rishijain — APPROVED** [review]

**Comment by KostiantynPopovych** [comment]
Hey @hmdros! May you have a quick look at this PR if you have some free time?

**Review by etagwerker — APPROVED** [review]
@sean-dickinson Looks good, thank you!
